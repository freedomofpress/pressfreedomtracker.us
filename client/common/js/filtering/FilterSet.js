import React, { PropTypes } from 'react'
import {
	AutocompleteInput,
	BoolInput,
	ChoiceInput,
	DateRangeInput,
	IntInput,
	TextInput,
	RadioPillInput,
} from '~/filtering/Inputs'

function FilterSet({ filters, handleFilterChange, filterValues }) {
	const components = filters.map((filter, index) => {
		if (filter.type === 'choice') {
			return (
				<ChoiceInput
					handleFilterChange={handleFilterChange}
					filterValues={filterValues}
					label={filter.title}
					filter={filter.name}
					choices={filter.choices}
					key={index}
				/>
			)
		} else if (filter.type === 'date') {
			return (
				<DateRangeInput
					handleFilterChange={handleFilterChange}
					filterValues={filterValues}
					label={filter.title}
					filter={filter.name}
					key={index}
				/>
			)
		} else if (filter.type === 'text') {
			return (
				<TextInput
					handleFilterChange={handleFilterChange}
					filterValues={filterValues}
					label={filter.title}
					filter={filter.name}
					key={index}
				/>
			)
		} else if (filter.type === 'radio') {
			return (
				<RadioPillInput
					handleFilterChange={handleFilterChange}
					filterValues={filterValues}
					label={filter.title}
					filter={filter.name}
					key={index}
				/>
			)
		} else if (filter.type === 'bool') {
			return (
				<BoolInput
					handleFilterChange={handleFilterChange}
					filterValues={filterValues}
					label={`${filter.title}`}
					filter={filter.name}
					key={index}
				/>
			)
		} else if (filter.type === 'int') {
			return (
				<IntInput
					handleFilterChange={handleFilterChange}
					filterValues={filterValues}
					label={`${filter.title}`}
					filter={filter.name}
					key={index}
					units={filter.units}
				/>
			)
		} else if (filter.type === 'autocomplete') {
			return (
				<AutocompleteInput
					handleFilterChange={handleFilterChange}
					filterValues={filterValues}
					label={filter.title}
					filter={filter.name}
					type={filter.autocomplete_type}
					isSingle={!filter.many}
					key={index}
				/>
			)
		}
		console.error('Unknown filter type.')
		return null
	})
	return (
		<div className="filters__set">
			{components}
		</div>
	)
}


FilterSet.propTypes = {
	filters: PropTypes.array.isRequired,
	handleFilterChange: PropTypes.func.isRequired,
	filterValues: PropTypes.object.isRequired,
}


export default FilterSet
