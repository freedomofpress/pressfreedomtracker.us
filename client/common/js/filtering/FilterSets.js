import React from 'react'
import {
	AutocompleteInput,
	BoolInput,
	ChoiceInput,
	DateRangeInput,
	TextInput,
	RadioPillInput,
} from '~/filtering/Inputs'


function FilterSet({ children }) {
	return (
		<div className="filters__set">
			{children}
		</div>
	)
}


const FilterSets = {}


FilterSets['General'] = function({ handleFilterChange, filterValues }) {
	return (
		<FilterSet>
			<TextInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Search Terms"
				filter="search"
			/>

			<DateRangeInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Took place between"
				filter="date"
			/>

			<TextInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Affiliation"
				filter="affiliation"
			/>

			<TextInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="City"
				filter="city"
			/>

			<AutocompleteInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="State"
				filter="state"
				type="incident.State"
				isSingle={true}
			/>

			<AutocompleteInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Targeted any of these journalists"
				filter="targets"
				type="incident.Target"
				isSingle={false}
			/>

			<AutocompleteInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Has any of these tags"
				filter="tags"
				type="common.CommonTag"
				isSingle={false}
			/>

			<TextInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Lawsuit Name"
				filter="lawsuit_name"
			/>

			<AutocompleteInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Venue"
				filter="venue"
				type="incident.Venue"
				isSingle={true}
			/>
		</FilterSet>
	)
}


FilterSets['General'].fields = [
	'search',
	'date_lower',
	'date_upper',
	'affiliation',
	'city',
	'state',
	'targets',
	'tags',
	'lawsuit_name',
	'venue',
]


const arrestSets = function({ handleFilterChange, filterValues, choices }) {
	return (
		<FilterSet>
			<BoolInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Unnecessary use of force?"
				filter="unnecessary_use_of_force"
			/>

			<ChoiceInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Arrest status"
				filter="arrest_status"
				choices={choices.ARREST_STATUS}
			/>

			<ChoiceInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Status of charges"
				filter="status_of_charges"
				choices={choices.STATUS_OF_CHARGES}
			/>

			<AutocompleteInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Charges"
				filter="charges"
				type="incident.Charge"
				isSingle={false}
			/>

			<DateRangeInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Detention date between"
				filter="detention_date"
			/>

			<DateRangeInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Release date between"
				filter="release_date"
			/>
		</FilterSet>
	)
}

FilterSets['Arrest / Detention'] = FilterSets['Arrest/Criminal Charge'] = FilterSets['Arrest / Criminal Charge'] = arrestSets

arrestSets.fields = [
	'unnecessary_use_of_force',
	'arrest_status',
	'status_of_charges',
	'charges',
	'detention_status',
	'detention_date_lower',
	'detention_date_upper',
	'release_date_lower',
	'release_date_upper',
]


FilterSets['Border Stop'] = function({ handleFilterChange, filterValues, choices }) {
	return (
		<FilterSet>
			<TextInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Border Point"
				filter="border_point"
			/>

			<BoolInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Stopped at border?"
				filter="stopped_at_border"
			/>

			<BoolInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Stopped previously?"
				filter="stopped_previously"
			/>

			<BoolInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Denied entry?"
				filter="denial_of_entry"
			/>

			<ChoiceInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="US Citizenship Status"
				filter="target_us_citizenship_status"
				choices={choices.CITIZENSHIP_STATUS_CHOICES}
			/>

			<RadioPillInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Did authorities ask for device access?"
				filter="did_authorities_ask_for_device_access"
			/>

			<RadioPillInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Did authorities ask for social media username?"
				filter="did_authorities_ask_for_social_media_user"
			/>

			<RadioPillInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Did authorities ask for social media password?"
				filter="did_authorities_ask_for_social_media_pass"
			/>

			<RadioPillInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Did authorities ask about work?"
				filter="did_authorities_ask_about_work"
			/>

			<RadioPillInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Were devices searched or seized?"
				filter="were_devices_searched_or_seized"
			/>

			<AutocompleteInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Nationality"
				filter="target_nationality"
				type="incident.Nationality"
				isSingle={false}
			/>
		</FilterSet>
	)
}

FilterSets['Border Stop'].fields = [
	'border_point',
	'stopped_at_border',
	'stopped_previously',
	'target_us_citizenship_status',
	'denial_of_entry',
	'target_nationality',
	'did_authorities_ask_for_device_access',
	'did_authorities_ask_for_social_media_user',
	'did_authorities_ask_for_social_media_pass',
	'did_authorities_ask_about_work',
	'were_devices_searched_or_seized',
]


FilterSets['Denial of Access'] = function({ handleFilterChange, filterValues }) {
	return (
		<FilterSet>
			<AutocompleteInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Politicians or public figures involved"
				filter="politicians_or_public_figures_involved"
				type="incident.PoliticianOrPublic"
				isSingle={false}
			/>
		</FilterSet>
	)
}

FilterSets['Denial of Access'].fields = [
	'politicians_or_public_figures_involved'
]


FilterSets['Equipment Search, Seizure, or Damage'] = function({ handleFilterChange, filterValues, choices }) {
	return (
		<FilterSet>
			<BoolInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Search warrant obtained?"
				filter="is_search_warrant_obtained"
			/>

			<ChoiceInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Status of seized equipment"
				filter="status_of_seized_equipment"
				choices={choices.STATUS_OF_SEIZED_EQUIPMENT}
			/>

			<ChoiceInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Actor"
				filter="actor"
				choices={choices.ACTORS}
			/>

			<AutocompleteInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Equipment seized includes any of"
				filter="equipment_seized"
				type="incident.Equipment"
				isSingle={false}
			/>

			<AutocompleteInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Equipment broken includes any of"
				filter="equipment_broken"
				type="incident.Equipment"
				isSingle={false}
			/>
		</FilterSet>
	)
}

FilterSets['Equipment Search, Seizure, or Damage'].fields = [
	'equipment_seized',
	'equipment_broken',
	'status_of_seized_equipment',
	'is_search_warrant_obtained',
	'actor',
]

FilterSets['Leak Case'] = function({ handleFilterChange, filterValues }) {
	return (
		<FilterSet>
			<BoolInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Charged under espionage act?"
				filter="charged_under_espionage_act"
			/>
		</FilterSet>
	)
}

FilterSets['Leak Case'].fields = [
	"charged_under_espionage_act"
]


FilterSets['Physical Attack'] = function({ handleFilterChange, filterValues, choices }) {
	return (
		<FilterSet>
			<RadioPillInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Was journalist targeted?"
				filter="was_journalist_targeted"
			/>

			<ChoiceInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Assailant"
				filter="assailant"
				choices={choices.ACTORS}
			/>
		</FilterSet>
	)
}

FilterSets['Physical Attack'].fields = [
	'assailant',
	'was_journalist_targeted',
]


FilterSets['Subpoena / Legal Order'] = function({ handleFilterChange, filterValues, choices }) {
	return (
		<FilterSet>
			<ChoiceInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Subpoena subject"
				filter="subpoena_subject"
				choices={choices.SUBPOENA_SUBJECT}
			/>

			<ChoiceInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Subpoena type"
				filter="subpoena_type"
				choices={choices.SUBPOENA_TYPE}
			/>

			<ChoiceInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Subpoena status"
				filter="subpoena_status"
				choices={choices.SUBPOENA_STATUS}
			/>

			<ChoiceInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Detention status"
				filter="detention_status"
				choices={choices.DETENTION_STATUS}
			/>

			<TextInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Third party in possession of communications"
				filter="third_party_in_possession_of_communications"
			/>

			<RadioPillInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Held in contempt?"
				filter="held_in_contempt"
			/>

			<ChoiceInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Third party business"
				filter="third_party_business"
				choices={choices.THIRD_PARTY_BUSINESS}
			/>

			<ChoiceInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Legal order type"
				filter="legal_order_type"
				choices={choices.LEGAL_ORDER_TYPES}
			/>

			<ChoiceInput
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
				label="Status of prior restraint"
				filter="status_of_prior_restraint"
				choices={choices.PRIOR_RESTRAINT_STATUS}
			/>
		</FilterSet>
	)
}

FilterSets['Subpoena / Legal Order'].fields = [
		'third_party_in_possession_of_communications',
		'third_party_business',
		'legal_order_type',
		'status_of_prior_restraint',
		'subpoena_subject',
		'subpoena_type',
		'subpoena_status',
		'held_in_contempt',
		'detention_status',
	]


export default FilterSets
