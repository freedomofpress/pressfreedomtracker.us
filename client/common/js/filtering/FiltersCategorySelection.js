import React, { PureComponent } from 'react'
import classNames from 'classnames'


class FiltersCategorySelection extends PureComponent {
	render() {
		const {
			categories,
			categoriesEnabled,
			handleSelection,
		} = this.props

		return (
			<div className="filters__category-selection">
				<div className="filters__text">
					Limit to
				</div>

				<ul className="filters__categories">
					{categories.map(category => (
						<li
							key={category.id}
							className={classNames(
								'filters__category',
								{ 'filters__category--active': categoriesEnabled[category.id] }
							)}
							onClick={handleSelection.bind(null, category.id)}
						>
							{category.title}
						</li>
					))}
				</ul>
			</div>
		)
	}
}


export default FiltersCategorySelection
