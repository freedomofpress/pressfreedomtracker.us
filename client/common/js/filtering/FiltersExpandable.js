import React from 'react'
import classNames from 'classnames'


function FiltersExpandable({ filtersExpanded, children }) {
	return (
		<div className={classNames(
			'filters__expandable',
			{ 'filters__expandable--expanded': filtersExpanded }
		)}>
			{children}
		</div>
	)
}


export default FiltersExpandable
