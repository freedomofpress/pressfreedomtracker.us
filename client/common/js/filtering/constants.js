export const HUMAN_DATE_FORMAT = 'MMM Do YYYY'
export const DATE_FORMAT = 'YYYY-MM-DD'
export const GENERAL_ID = -1
