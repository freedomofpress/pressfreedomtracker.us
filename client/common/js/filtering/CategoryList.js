import React from 'react'

import { GENERAL_ID } from '~/filtering/constants'


function CategoryList({ categories }) {
	return (
		<ul className="filters__summary-list">
			{categories.filter(({ id }) => id !== GENERAL_ID).map(category => (
				<li key={category.id} className="filters__summary-item filters__summary-item--semicolons">
					{category.title}
				</li>
			))}
		</ul>
	)
}


export default CategoryList
