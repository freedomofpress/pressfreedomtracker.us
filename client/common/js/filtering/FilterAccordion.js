import React from 'react'
import classNames from 'classnames'
import { CollapseIcon, ExpandIcon } from '~/filtering/Icons'
import FilterSets from '~/filtering/FilterSets'
import NewFilterSet from '~/filtering/NewFilterSet'


function FilterAccordion({
	category,
	selectedAccordions,
	handleAccordionSelection,
	handleFilterChange,
	filterValues,
	noCategoryFiltering,
	choices,
}) {
	if (!category.enabled) {
		return null
	}
	let FilterSet
	if(category.id === -1) {
		// General is a special category with id -1 that should be rendered
		FilterSet = typeof FilterSets[category.title] === 'function' ? FilterSets[category.title] : null
	} else if (!category.related_fields || !(category.related_fields.length > 0)) {
		// Don't bother rendering an accordion with no filters
		return null
	}


	const isActive = selectedAccordions.includes(category.id)
	const collapsible = !noCategoryFiltering

	let renderedFilterset
	if(isActive && category.related_fields) {
		renderedFilterset = (<NewFilterSet
			title={category.title}
			fields={category.related_fields}
			handleFilterChange={handleFilterChange}
			filterValues={filterValues}
			choices={choices}
			key={category.id}
		/>)
	} else if(isActive && FilterSet) {
		renderedFilterset = (
			<FilterSet
				handleFilterChange={handleFilterChange}
				filterValues={filterValues}
			/>
		)
	} else {
		renderedFilterset = null
	}

	return (
		<li
			className={classNames(
				'filters__accordion-category',
				{ 'filters__accordion-category--no-divider': noCategoryFiltering }
			)}
		>
			{collapsible && (
				<button
					className={classNames(
						'filters__accordion',
						{ 'filters__accordion--active': isActive }
					)}
					onClick={handleAccordionSelection.bind(null, category.id)}
					type="button"
				>
					{isActive ? <CollapseIcon /> : <ExpandIcon />}
					{category.title}
				</button>
			)}

			{renderedFilterset}
		</li>
	)
}


export default FilterAccordion
