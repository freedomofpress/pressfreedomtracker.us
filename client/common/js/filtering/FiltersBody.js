import React, { PureComponent } from 'react'
import FilterAccordion from '~/filtering/FilterAccordion'


class FiltersBody extends PureComponent {
	render() {
		const {
			selectedAccordions,
			handleAccordionSelection,
			categoriesEnabled,
			handleFilterChange,
			filterValues,
			noCategoryFiltering,
			choices,
		} = this.props
		return (
			<ul className="filters__body">
				<FilterAccordion
					category={{
						id: -1,
						title: 'General',
						enabled: true,
					}}
					handleAccordionSelection={handleAccordionSelection}
					handleFilterChange={handleFilterChange}
					selectedAccordions={selectedAccordions}
					filterValues={filterValues}
					noCategoryFiltering={noCategoryFiltering}
					choices={choices}
				/>

				{categoriesEnabled.map(category => (
					<FilterAccordion
						key={category.id}
						category={category}
						handleAccordionSelection={handleAccordionSelection}
						handleFilterChange={handleFilterChange}
						selectedAccordions={selectedAccordions}
						filterValues={filterValues}
						noCategoryFiltering={noCategoryFiltering}
						choices={choices}
					/>
				))}
			</ul>
		)
	}
}


export default FiltersBody
