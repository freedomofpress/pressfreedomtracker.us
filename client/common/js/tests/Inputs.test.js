import React from 'react';
import axios from 'axios';
import { shallow, mount } from 'enzyme';
import { AutocompleteInput } from '~/filtering/Inputs';
import Autocomplete from 'WagtailAutocomplete/Autocomplete';
import { TAG_FILTER_VALUES, EMPTY_FILTER_VALUES, TAG_MOCK_RESULTS } from '~/tests/factories/CategoriesFilterFactory';

jest.mock('axios');

// Allows me to ensure that all the promises are resolved.
// setImmediate isn't to be used in production but should
// be fine for testing and CI.
const flushPromises = () => new Promise(setImmediate);

describe('Inputs', () => {

    it('should render AutocompleteInput without error in "debug" mode', () => {
        const component = shallow(
            <AutocompleteInput
                handleFilterChange={jest.fn()}
                filter={'tags'}
                filterValues={EMPTY_FILTER_VALUES}
                label={'Autocomplete'}
                isSingle={false}
                type={'common.CommonTag'}
            />
        );

        expect(component.debug()).toMatchSnapshot();
    });

	it('should render Autocomplete correctly', async () => {
        axios.get.mockImplementationOnce(() => Promise.resolve(
            {
                'data': TAG_MOCK_RESULTS,
                'status': 200
            }
        ));
		const component = mount(
            <AutocompleteInput
                handleFilterChange={jest.fn()}
                filter={'tags'}
                filterValues={EMPTY_FILTER_VALUES}
                label={'Autocomplete'}
                isSingle={false}
                type={'common.CommonTag'}
            />
        )
        await flushPromises();
        component.update();
        const autocompleteComponent = component.find('.autocomplete');
        expect(autocompleteComponent.length).toEqual(1);
        expect(autocompleteComponent.find('.suggestions .suggestions__item').length).toEqual(2);
        component.find('.autocomplete__search').simulate('focus');
        expect(component.find('.suggestions').prop('style')).toHaveProperty('display', 'block');
    });

    it('should render Autocomplete selection results correctly when clicked', async() => {
        axios.get.mockImplementationOnce(() => Promise.resolve(
            {
                'data': TAG_MOCK_RESULTS,
                'status': 200
            }
        ));
		const component = mount(
            <AutocompleteInput
                handleFilterChange={jest.fn()}
                filterValues={EMPTY_FILTER_VALUES}
                label={'Autocomplete'}
                filter={'tags'}
                isSingle={false}
                type={'common.CommonTag'}
            />
        );

        // Mock function to update the props of the root component
        const handleFilterChangeMock = jest.fn((label, event) => {
            component.setProps({filterValues: {[label]: event.target.value}});
        });
        component.setProps({handleFilterChange: handleFilterChangeMock});

        // Used to complete all the promises
        await flushPromises();
        component.update();

        // Before clicking nothing exist
        expect(component.find('.autocomplete-layout__item--padded span').text()).toEqual('Nothing selected.');
        component.find('.autocomplete__search').simulate('focus');

        // Simulate clicking the first option
        const suggestions = component.find('.suggestions .suggestions__item');
        suggestions.first().simulate('click');
        component.update();

        // Check clicking renders the selection
        const selection = component.find('.autocomplete-layout__item--padded .selection');
        expect(selection.length).toEqual(1);
        expect(selection.find('.selection__label').text()).toEqual('Tag 1');

        // Simulate clicking the apply filter button by setting props
        const fetchResult = {
            items: TAG_MOCK_RESULTS.items.filter(result => result.pk === TAG_FILTER_VALUES.tags[0].id),
        }
        axios.get.mockImplementation(() => Promise.resolve(
            {
                'data': fetchResult,
                'status': 200
            }
        ));
        component.setProps({filterValues: TAG_FILTER_VALUES});
        // Used to complete all the promises
        await flushPromises();
        component.update();
        // Check selection still has text
        expect(component.find('.selection .selection__label').text()).toEqual('Tag 1');
    });

    it('should render Autocomplete correctly when filtervalues are there in URL', async () => {
        const fetchResult = {
            items: TAG_MOCK_RESULTS.items.filter(result => result.pk === TAG_FILTER_VALUES.tags[0].id),
        }
        axios.get.mockImplementation((url) => {
            if (url.indexOf('/autocomplete/search/') !== -1) {
                return Promise.resolve(
                    {
                        'data': TAG_MOCK_RESULTS,
                        'status': 200
                    }
                )
            } else if (url.indexOf('/autocomplete/objects/') !== -1) {
                return Promise.resolve(
                    {
                        'data': fetchResult,
                        'status': 200
                    }
                )
            }
        });
		const component = mount(
            <AutocompleteInput
                handleFilterChange={jest.fn()}
                filter={'tags'}
                filterValues={TAG_FILTER_VALUES}
                label={'Autocomplete'}
                isSingle={false}
                type={'common.CommonTag'}
            />
        );
        // Mock function to update the props of the root component
        const handleFilterChangeMock = jest.fn((label, event) => {
            component.setProps({filterValues: {[label]: event.target.value}});
        });
        component.setProps({handleFilterChange: handleFilterChangeMock});

        await flushPromises();
        component.update();
        const selection = component.find('.autocomplete-layout__item--padded .selection');
        expect(selection.length).toEqual(1);
        expect(selection.find('.selection__label').text()).toEqual('Tag 1');
    });
});
