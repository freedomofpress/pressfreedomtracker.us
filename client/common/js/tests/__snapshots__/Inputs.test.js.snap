// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inputs should render AutocompleteInput without error in "debug" mode 1`] = `
"<div className=\\"filters__input-row\\">
  <span className=\\"filters__input-label\\">
    Autocomplete
  </span>
  <Autocomplete type=\\"common.CommonTag\\" name=\\"tags\\" canCreate={false} isSingle={false} value={[undefined]} onChange={[Function: bound mockConstructor]} fetchInitialValues={true} apiBase=\\"/autocomplete/\\" controlled={true} />
</div>"
`;
